### DISCUSSION TITLE
[MQTT] Integrazione shelly tenda da sole

### DISCUSSION THREAD

**Discussion by baldus88** [indomus/forum#407] [Hardware & Firmware]
Ciao.
Da poco ho collegato 4 shelly per le tende da sole del terrazzo. Ci sono un paio che sono a segnale pieno altre due più faticoso:

![image](https://user-images.githubusercontent.com/57867040/121354219-f752a800-c92e-11eb-89c3-b6c1864e0309.png)

![image](https://user-images.githubusercontent.com/57867040/121354249-fcaff280-c92e-11eb-8820-41e8b7bd339d.png)

Ho inizialmente usato l'integrazione nativa di shelly per comodità e ogni tanto vedo che spariscono totalmente da home assistant pure quelle che prendono bene.

![image](https://user-images.githubusercontent.com/57867040/121355211-ef473800-c92f-11eb-90db-46dccc518faf.png)

Ho fatto un test ad integrarne una manualmente tramite mqtt ma quando provo a muoverla mi va in unavailable:

![image](https://user-images.githubusercontent.com/57867040/121354605-52849a80-c92f-11eb-84fd-d2e01613a7e7.png)

Potrebbe essere un discorso di configurazione o proprio di connettività?

Questa è la mia configurazione:

- platform: mqtt
  name: Tenda Interno Sinistro 
  command_topic: "shellies/shellyswitch25-xx/roller/0/command"
  position_topic: "shellies/shellyswitch25-xx/roller/0/pos"
  set_position_topic: "shellies/shellyswitch25-xx/roller/0/command/pos"
  availability_topic: "shellies/shellyswitch25-xx/online"
  payload_available: "true"
  payload_not_available: "false"
  qos: 1
  retain: false
  payload_open: "open"
  payload_close: "close"
  payload_stop: "stop"
  position_open: 100
  position_closed: 0
  optimistic: false

**Comment by snakuzzo** [comment]
perché se si scollega dalla WIFI, si scollega anche dal broker MQTT ed è giusto che ti dia `unavailable`
E' tutto basato sull'`availability_topic` basato sul [LWT](https://indomus.it/formazione/mqtt-cose-e-come-funziona-il-last-will-and-testament-lwt/)

**Comment by baldus88** [comment]
Sono d'accordo con te se si disconnettesse totalmente, solamente che se non faccio niente in quel momento non cade, se provo a muovere lo slider va in unavailable, peccato che se vado ad aprire l'entità fornita dall'integrazione nativa me lo vede ancora su, quindi non cade nella casisistica che si sia sconnesso del tutto. Non so se mi sono spiegato.

**Reply by snakuzzo** [reply]
Allora hai sbagliato qualcosa nella creazione dell'entità MQTT. Guarda quello che abbiamo scritto in guida e controlla bene topic e payload. Vedi anche cosa accade sulla coda del broker mettendoti in ascolto
